### DISCUSSION TITLE
$25: Webcam Selection

### DISCUSSION THREAD

**Discussion by hacksider** [hacksider/Deep-Live-Cam#673] [Bounty]
Ok, here's a simple task for everyone (just making this a fun game for the community)! There's already a commit of this on the experimental. Merge this (incorporate the feature) to the main and recommit it. Then just push it in the bounty branch! Please make sure it works on your end _**without breaking any current features**_

![image](https://github.com/user-attachments/assets/2126cac0-06d4-4f0b-af4c-d0690a76b32a)

Please include the word "BOUNTY - Webcam Merged (tested)" on your pull request.

Bounty: $25 (Paid via Paypal)

Edit: Just making the instructions clearer
Edit 2: Making the bounty $25 instead of $20

**Comment by AbandonedCart** [comment]
Anyone in need of a hint should check out https://github.com/hacksider/Deep-Live-Cam/pull/115

**Reply by AbandonedCart** [reply]
Hint 2 - Pay attention to how the UI changed in https://github.com/hacksider/Deep-Live-Cam/commits/main/modules/ui.py

**Comment by negativenagesh** [comment]
@hacksider @AbandonedCart 

# So is this works like- the first person to complete the task gets the bounty or anyone/everyone who completes?

**Reply by AbandonedCart** [reply]
I honestly don’t know the rules (and apologize if giving a hint broke any of them). I just wanted to give those with less experience a better chance.

**Reply by hacksider** [reply]
> @hacksider @AbandonedCart
> 
> # So is this works like- the first person to complete the task gets the bounty or anyone/everyone who completes?

![image](https://github.com/user-attachments/assets/197cdc0c-d0e5-4a9d-b543-f3da687a9e02)

**Reply by hacksider** [reply]
There will still be some QA to ensure quality. If it reached the main branch, I will release the bounty. Of course this is just a gamification of some task. Everyone will still benefit on this as this is an open source app (and also considered a contributor as well).

**Comment by negativenagesh** [comment]
@hacksider , I have a doubt....

should I merge only ui.py from experimental to main or all the files?

**Reply by hacksider** [reply]
Add the feature webcam selection on the current commit without breaking it up. You should test it works on your end before pushing the commit. 😄

**Comment by hacksider** [comment]
I added more info on the top (some edits) to make sure you just don't merge and commit :)

**Comment by KRSHH** [comment]
ATTENTION: Add the Webcam selection on top of the latest commit - [d4e5b80](https://github.com/hacksider/Deep-Live-Cam/commit/d4e5b8078d8cd63b5c5b16d534db26ccd41b29db).

**Comment by ByteJoseph** [comment]
The bounty branch is 7 commits behind. What should I do next ?

**Reply by hacksider** [reply]
Just commit to the main :)

**Reply by hacksider** [reply]
Notification: I just updated the bounty to the latest commit

**Reply by hacksider** [reply]
You can commit on the bounty again

**Comment by hacksider** [comment]
closed, https://github.com/hacksider/Deep-Live-Cam/pull/686

**Comment by hacksider** [comment]
Awarded to @Ghoulboii
